performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/Process.swift

```diff
@@ -0,0 +1,217 @@
+//
+//  Copyright 2021 Google LLC
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import SwiftUI
+import FirebasePerformance
+import Vision
+
+class Process: ObservableObject {
+  @Published var status: ProcessStatus = .idle
+  @Published var action: ProcessAction?
+  @Published var image: UIImage?
+  var categories: [(category: String, confidence: VNConfidence)] = []
+  let precision: Float = 0.7
+  let trace = Performance.sharedInstance().trace(name: "Classification")
```

Comment:
This is only used in one function, so instead of creating it on initialization and retaining it you should create and use it in the function itself. This will prevent issues like accidentally reusing traces that have already been stopped (for example if you call `downloadImageAsync()` twice).

Reply:
Done ✅